integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/putmessage/PostVirksomhetStrategyFactory.java

```diff
@@ -16,15 +19,25 @@ private PostVirksomhetStrategyFactory(CorrespondenceAgencyConfiguration configur
 
     public static PostVirksomhetStrategyFactory newInstance(IntegrasjonspunktProperties properties,
                                                             ServiceRegistryLookup serviceRegistryLookup) {
-        return new PostVirksomhetStrategyFactory(
-                new CorrespondenceAgencyConfiguration.Builder()
+        CorrespondenceAgencyConfiguration.Builder builder = new CorrespondenceAgencyConfiguration.Builder()
                 .withExternalServiceCode(properties.getAltinnPTV().getExternalServiceCode())
                 .withExternalServiceEditionCode(properties.getAltinnPTV().getExternalServiceEditionCode())
                 .withPassword(properties.getAltinnPTV().getPassword())
-                .withSystemUserCode(properties.getAltinnPTV().getUsername())
-                .build(),
-                serviceRegistryLookup
-        );
+                .withSystemUserCode(properties.getAltinnPTV().getUsername());
+
+        Optional<String> smsVarslingstekst = Optional.of(properties.getAltinnPTV()).map(IntegrasjonspunktProperties
+                .PostVirksomheter::getSms).map(IntegrasjonspunktProperties.Sms::getVarslingstekst);
+        if (smsVarslingstekst.isPresent() && !"".equals(smsVarslingstekst.get())) {
+            builder.withSmsText(smsVarslingstekst.get());
```

Comment:
Det var mye kode i PTV factory for å hente ut en property. Men det ser jo gøy ut da.